src/test/java/school/hei/tsinjo/service/mapper/VolaPspTest.java

```diff
@@ -0,0 +1,74 @@
+package school.hei.tsinjo.service.mapper;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.time.Instant;
+import java.util.Date;
+import org.junit.jupiter.api.Test;
+import school.hei.tsinjo.model.Payment;
+import school.hei.tsinjo.model.psp.PspType;
+import school.hei.tsinjo.model.psp.vola.VolaPsp;
+import school.hei.tsinjo.model.psp.vola.api.VolaClient;
+import school.hei.tsinjo.model.psp.vola.api.gen.client.model.PspPayment;
+
+public class VolaPspTest {
+
+  @Test
+  void creationInstant_should_be_mapped_from_pspPayment() {
+    String tsinjoId = "tsinjo-test";
+    String pspId = "MP250805.0922.B95953";
+    String email = "ninah@mail.hei.school";
+
+    var rawPayment = createRawPaymentWithPspPayment(pspId);
+    VolaClient volaClient = mock(VolaClient.class);
+    when(volaClient.get(PspType.ORANGE_MONEY, pspId, email)).thenReturn(rawPayment);
+
+    VolaPsp volaPsp = new VolaPsp(volaClient);
+    Payment mapped = volaPsp.get(tsinjoId, PspType.ORANGE_MONEY, pspId, email);
+
+    assertMappedPayment(mapped, pspId);
+  }
+
+  private school.hei.tsinjo.model.psp.vola.api.gen.client.model.Payment
+      createRawPaymentWithPspPayment(String pspId) {
+    var raw = new school.hei.tsinjo.model.psp.vola.api.gen.client.model.Payment();
+    raw.setId("vola-raw-id-1");
+
+    var psp = createPspPayment(pspId);
+    raw.setPspPayment(psp);
+
+    raw.setCreationInstant(Date.from(Instant.parse("2025-09-04T14:45:44.142Z")));
+    raw.setLastPspVerificationInstant(Date.from(Instant.parse("2025-09-05T08:22:02.683Z")));
+    raw.setVerificationStatus(
+        school.hei.tsinjo.model.psp.vola.api.gen.client.model.Payment.VerificationStatusEnum
+            .SUCCEEDED);
+
+    return raw;
+  }
+
+  private PspPayment createPspPayment(String pspId) {
+    var psp = new PspPayment();
+    psp.setPspType(PspPayment.PspTypeEnum.ORANGE_MONEY);
+    psp.setId(pspId);
+    psp.setAmount(3000);
+    Instant expectedPspCreation = Instant.parse("2025-09-04T17:41:55Z");
+    psp.setCreationInstant(Date.from(expectedPspCreation));
+    return psp;
+  }
+
+  private void assertMappedPayment(Payment mapped, String pspId) {
```

Comment:
find a betterName. If I just read the method name, I have no idea of what you're doing